tests/framework/extensions/rke1/nodepools/nodepools.go

```diff
@@ -7,9 +7,14 @@ import (
 	"github.com/rancher/norman/types"
 	"github.com/rancher/rancher/tests/framework/clients/rancher"
 	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
+const (
+	active = "active"
+)
+
 // IsNodeReady is a helper method that will loop and check if the node is ready in the RKE1 cluster.
 // It will return an error if the node is not ready after set amount of time.
 func IsNodeReady(client *rancher.Client, nodePool *management.NodePool, ClusterID string) error {
```

Comment:
yeah I had noted that sometime back as well. When I had removed this function in favor of the other, the nodes didn't properly wait. I'm pretty sure it was because the v3 node pool wasn't being accounted for so it just assumed all the nodes were good to go.

I was thinking of future test scenarios if we wanted to not have only RKE1 and that may provide issues. However, if we think that only we will need the one function, I can add the v3 node pool parameter and remove the `nodepools.go` version no problem.

Reply:
So if I get it correctly, does this mean the old one works for RKE1, whereas the new one works for all types of clusters - including v1 clusters?

Reply:
That is correct. Israel made a point I just saw of using Steve, I can try to adjust this cause I agree, I would like to remove redundancy where possible.

Reply:
As mentioned in another comment, I can actually use Management client for both. I updated `client.Management.Node.List` to `client.Management.Node.ListAll` and validated this works for all cluster types.

Reply:
Resolved.